### PR TITLE
RAP-4062 - fix bug related to doc/series repo deletion

### PR DESCRIPTION
**Pull request by dukenguyen**

### CODE REVIEW

File: Libs/RaptureCore/src/main/java/rapture/kernel/SearchApiImpl.java [hunk 1]
```diff
@@ -60,19 +59,19 @@ public SearchApiImpl(Kernel raptureKernel) {
 
     // Trusted calls
 
-    public void writeSearchEntry(String repo, DocumentWithMeta doc) {
-        logger.info("Writing doc search entry to " + repo);
-        Kernel.getRepoCacheManager().getSearchRepo(repo).put(doc);
+    public void writeSearchEntry(String searchRepo, DocumentWithMeta doc) {
+        logger.info(String.format("Writing doc search entry to [%s]", searchRepo));
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SearchApiImpl.java]: debug level?

File: Libs/RaptureCore/src/main/java/rapture/kernel/pipeline/SearchPublisher.java [hunk 2]
```diff
@@ -71,48 +71,46 @@ public static void publishDeleteMessage(CallingContext context, Searchable searc
...
         Kernel.getPipeline().publishMessageToCategory(context, task);
     }
 
     private static boolean shouldPublish(Searchable searchableRepo, String uri) {
         if (ConfigLoader.getConf().FullTextSearchOn && searchableRepo.getFtsIndex()) {
-            if (StringUtils.isBlank(searchableRepo.getFtsIndexRepo())) {
-                searchableRepo.setFtsIndexRepo(ConfigLoader.getConf().FullTextSearchDefaultRepo);
-            }
-            log.info(String.format("Publishing search update for uri [%s] to search repo [%s] ...", uri, searchableRepo.getFtsIndexRepo()));
+            log.info(String.format("Publishing search update for uri [%s] to search repo [%s] ...", uri, SearchRepoUtils.getSearchRepo(searchableRepo)));
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/kernel/pipeline/SearchPublisher.java]: change to debug level

File: Libs/RaptureCore/src/main/java/rapture/kernel/SearchApiImpl.java [hunk 3]
```diff
@@ -60,19 +59,19 @@ public SearchApiImpl(Kernel raptureKernel) {
...
+        Kernel.getRepoCacheManager().getSearchRepo(searchRepo).put(doc);
     }
 
-    public void writeSearchEntry(String repo, SeriesUpdateObject seriesUpdateObject) {
-        logger.info("Writing series search entry to " + repo);
-        Kernel.getRepoCacheManager().getSearchRepo(repo).put(seriesUpdateObject);
+    public void writeSearchEntry(String searchRepo, SeriesUpdateObject seriesUpdateObject) {
+        logger.info(String.format("Writing series search entry to [%s]", searchRepo));
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SearchApiImpl.java]: debug level

File: Libs/RaptureCore/src/main/java/rapture/kernel/pipeline/SearchPublisher.java (hunk 2 above)
Comment (dukenguyen) [inline Libs/RaptureCore/src/main/java/rapture/kernel/pipeline/SearchPublisher.java]: done.

File: Libs/RaptureCore/src/main/java/rapture/kernel/SearchApiImpl.java (hunk 3 above)
Comment (dukenguyen) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SearchApiImpl.java]: done

**Pull request merged**